dace/libraries/blas/nodes/axpy.py

```diff
@@ -0,0 +1,281 @@
+
+        vecAdd_sdfg = dace.SDFG('vecAdd_graph')
+        vecAdd_state = vecAdd_sdfg.add_state()
+
+        vecAdd_sdfg.add_symbol(a.name, dtype)
+
+        # ---------- ----------
+        # MEMORY LOCATIONS
+        # ---------- ----------
+        # vecAdd_sdfg.add_scalar('_a', dtype=dtype, storage=dtypes.StorageType.FPGA_Global)
+
+        x_in = vecAdd_state.add_stream('_x',
+                                       vecType,
+                                       buffer_size=32,
+                                       storage=dtypes.StorageType.FPGA_Local)
+        y_in = vecAdd_state.add_stream('_y',
+                                       vecType,
+                                       buffer_size=32,
+                                       storage=dtypes.StorageType.FPGA_Local)
+        z_out = vecAdd_state.add_stream('_res',
+                                        vecType,
+                                        buffer_size=32,
+                                        storage=dtypes.StorageType.FPGA_Local)
+
+        # ---------- ----------
+        # COMPUTE
+        # ---------- ----------
+        vecMap_entry, vecMap_exit = vecAdd_state.add_map(
+            'vecAdd_map',
+            dict(i='0:{0}/{1}'.format(n, vecWidth)),
+            schedule=dtypes.ScheduleType.FPGA_Device)
+
+        vecAdd_tasklet = vecAdd_state.add_tasklet(
+            'vecAdd_task', ['x_con', 'y_con'], ['z_con'],
+            'z_con = {} * x_con + y_con'.format(a))
+
+        vecAdd_state.add_memlet_path(x_in,
+                                     vecMap_entry,
+                                     vecAdd_tasklet,
+                                     dst_conn='x_con',
+                                     memlet=dace.Memlet.simple(
+                                         x_in.data, '0'))
+
+        vecAdd_state.add_memlet_path(y_in,
+                                     vecMap_entry,
+                                     vecAdd_tasklet,
+                                     dst_conn='y_con',
+                                     memlet=dace.Memlet.simple(
+                                         y_in.data, '0'))
+
+        vecAdd_state.add_memlet_path(vecAdd_tasklet,
+                                     vecMap_exit,
+                                     z_out,
+                                     src_conn='z_con',
+                                     memlet=dace.Memlet.simple(
+                                         z_out.data, '0'))
+
+        return vecAdd_sdfg
+
+    @staticmethod
+    def expansion(node, state, sdfg):
+        node.validate(sdfg, state)
+        if node.dtype is None:
+            raise ValueError("Data type must be set to expand " + str(node) +
+                             ".")
+        return ExpandAxpyFPGAStreaming.make_sdfg(node.dtype, int(node.vecWidth),
+                                                 node.n, node.a)
+
+
+
+@dace.library.node
+class Axpy(dace.sdfg.nodes.LibraryNode):
```

Comment:
Please add a docstring briefly explaining what this is (e.g., `Implements the BLAS AXPY operation, which is the sum of a vector...`).